ingestr/src/factory_test.py

```diff
@@ -11,3 +11,88 @@ def test_scheme_is_parsed_from_uri_correctly():
         parse_scheme_from_uri("redshift+psycopg2://user:pw@host") == "redshift+psycopg2"
     )
     assert parse_scheme_from_uri("mysql+pymysql://user:pw@host") == "mysql+pymysql"
+
+
+def test_parse_columns_valid_input():
+    input_columns = [
+        "col1:text",
+        "col2:double:nullable",
+        "col3:bool"
+    ]
+    expected_output = {
+        "col1": {"data_type": "text", "nullable": False},
+        "col2": {"data_type": "double", "nullable": True},
+        "col3": {"data_type": "bool", "nullable": False}
+    }
+    assert parse_columns(input_columns) == expected_output
+
+def test_parse_columns_empty_input():
+    assert parse_columns([]) == {}
+
+def test_parse_columns_none_input():
+    assert parse_columns(None) == {}
+
+def test_parse_columns_all_data_types():
+    input_columns = [
+        "col1:text", "col2:double", "col3:bool", "col4:timestamp",
+        "col5:bigint", "col6:binary", "col7:complex", "col8:decimal",
+        "col9:wei", "col10:date", "col11:time"
+    ]
+    result = parse_columns(input_columns)
+    assert len(result) == 11
+    for col, props in result.items():
+        assert props["nullable"] == False
```

Comment:
let's test for the array you expect here right away, we should expect the value to be exactly equal to what you expect.